### PR TITLE
Add potential for two-power density profile

### PR DESCRIPTION
**Pull request by adrn**
This adds an implementation of the potential for a two-power density profile (e.g., Eq. 2.64 in B&T08). I desperately wanted to avoid using the 2F1 hypergeometric function because, even though there is now an implementation in jax.scipy.special (jax-ml/jax#28168), it doesn't seem stable for x>~1 (which, in the potential, is r>~r_s -- a regime we clearly want to support!). So this implementation is restricted to cases when alpha < 2 and beta > 3 (i.e. it does not include beta == 3, the "generalized NFW" potential). As an aside, I noticed we have an implementation of gNFW, but it uses the tensorflow_probability implementation of 2F1, which seems to actually bork for x>1. 

cc @jnibauer @sapearson

### CODE REVIEW

**Comment by adrn** [comment]
Actually I have a working implementation of the Zhao (alpha, beta, gamma) double power-law model, which contains the two-power model, so I'm going to close this in favor of the incoming PR!

**Comment by jnibauer** [comment]
Sorry missed this earlier. On the Zhao double power-law model – are force evaluations super slow using this model? I had tried to implement it but gradients of the incomplete Beta-function were slow...
